test/functional/api/v1/bibs_controller_test.rb

```diff
@@ -1,17 +1,1351 @@
+          {
+            "tag": "a",
+            "content": "Books for Reading and Sharing - Elementary School!"
+          }
+        ]
+      },
+      {
+        "fieldTag": "u",
+        "marcTag": "246",
+        "ind1": "3",
+        "ind2": "0",
+        "content": nil,
+        "subfields": [
+          {
+            "tag": "a",
+            "content": "English Language Arts: Books for Reading and Sharing - Elementary School! Gr. 1-2 (Teacher Set)."
+          }
+        ]
+      },
+      {
+        "fieldTag": "y",
+        "marcTag": "901",
+        "ind1": " ",
+        "ind2": " ",
+        "content": nil,
+        "subfields": [
+          {
+            "tag": "a",
+            "content": "ed"
+          },
+          {
+            "tag": "b",
+            "content": "SEL"
+          }
+        ]
+      },
+      {
+        "fieldTag": "y",
+        "marcTag": "944",
+        "ind1": " ",
+        "ind2": " ",
+        "content": nil,
+        "subfields": [
+          {
+            "tag": "a",
+            "content": "9781896580609  9781442429789  9781484730881"
+          }
+        ]
+      },
+      {
+        "fieldTag": "_",
+        "marcTag": nil,
+        "ind1": nil,
+        "ind2": nil,
+        "content": "00000nam  2200000 a 4500",
+        "subfields": nil
+      }
+    ]
+  }
+]
+
+TWO_TEACHER_SETS_TO_DELETE = [{
+    "id": "0",
+    "title": "Example for non-existant bnumber"
+  },
+  {
+      "id": "#{BNUMBER1}",
+      "nyplSource": "sierra-nypl",
+      "nyplType": "bib",
+      "updatedDate": "2017-08-25T06:32:01-04:00",
+      "createdDate": nil,
+      "deletedDate": "2012-06-08",
+      "deleted": true,
+      "locations": [],
+      "suppressed": nil,
+      "lang": nil,
+      "title": nil,
+      "author": nil,
+      "materialType": nil,
+      "bibLevel": nil,
+      "publishYear": nil,
+      "catalogDate": nil,
+      "country": nil,
+      "normTitle": nil,
+      "normAuthor": nil,
+      "standardNumbers": [],
+      "controlNumber": "",
+      "fixedFields": {},
+      "varFields": [],
+      "count": 1,
+      "totalCount": 0,
+      "statusCode": 200,
+      "debugInfo": []
+    }
+]
+
 class Api::BibsControllerTest < ActionController::TestCase
   def setup
     @controller = Api::V1::BibsController.new
   end
 
-  test "should create or update teacher sets when given a bib number" do
-    post 'create_or_update_teacher_sets'
+  # test 'respond with 400 if request body is missing' do
+  #   post 'create_or_update_teacher_sets'
+  #   assert response.status == 400
+  # end
+
+  test "respond with 400 if ID isn't present for at least one JSON record" do
+    post 'create_or_update_teacher_sets', { _json: [{ not_an_id: 0 }] }
+    assert response.status == 400
+  end
+
+  test "should create or update teacher sets when given a ID (a.k.a. bnumber)" do
+    post 'create_or_update_teacher_sets', { _json: [{ id: BNUMBER1, another_field: 'x' }, { id: BNUMBER2, another_field: 'x' }] }
```

Comment:
would this fail without a title now?

Reply:
Yes, in the next PR we'll include validation.